### ISSUE TITLE
admonition support: warnings/hints/notes, etc.

### ISSUE THREAD

**Issue by skshetry** [iterative/dvc.org#3153]
Having a proper visually distinguishable box for admonitions would make it easier to read the documentations. Right now, we misuse markdown blockquotes and (sometime) use emojis for variants. But they are mixed in between other documentation paragraphs, that it is quite easy to miss (and does not look good to be honest).

Some screenshots where I found it confusing:
<img width="709" alt="Screen Shot 2022-01-06 at 16 19 59" src="https://user-images.githubusercontent.com/18718008/148369939-932bb8f5-92c9-453d-8ff5-9f52b99f294d.png">

<img width="709" alt="Screen Shot 2022-01-06 at 16 19 26" src="https://user-images.githubusercontent.com/18718008/148369953-2d807408-91bc-4594-b66a-a3beeea08f5c.png">
<img width="725" alt="Screen Shot 2022-01-06 at 16 19 12" src="https://user-images.githubusercontent.com/18718008/148369960-5c8a29c0-14f7-4490-a455-9f42120f48d8.png">
<img width="725" alt="Screen Shot 2022-01-06 at 16 19 02" src="https://user-images.githubusercontent.com/18718008/148369972-226f480b-8caf-4b7d-aff9-f1126fcb9842.png">

These are only a few examples that I quickly scanned through.

We can take some inspirations from spacy's and thinc's docs:
<img width="963" alt="Screen Shot 2022-01-06 at 16 23 30" src="https://user-images.githubusercontent.com/18718008/148370145-129d5bff-6363-4cad-8f69-bc2b53437c24.png">

<img width="963" alt="Screen Shot 2022-01-06 at 16 25 55" src="https://user-images.githubusercontent.com/18718008/148370668-dca4df32-b8bc-4a46-b9b4-103de7e0e126.png">

<img width="963" alt="Screen Shot 2022-01-06 at 16 25 31" src="https://user-images.githubusercontent.com/18718008/148371315-14be9837-468a-455d-8f26-51782eebf47c.png">

<img width="963" alt="Screen Shot 2022-01-06 at 16 24 29" src="https://user-images.githubusercontent.com/18718008/148371342-0b507227-d60b-49fc-8791-83ffa57f0a6f.png">

<img width="963" alt="Screen Shot 2022-01-06 at 16 26 51" src="https://user-images.githubusercontent.com/18718008/148370654-1d90ddae-5a25-41b4-8fe5-8fe62b46c412.png">

<img width="1042" alt="Screen Shot 2022-01-06 at 16 27 52" src="https://user-images.githubusercontent.com/18718008/148370604-60980931-3c0c-4c98-b6ea-c180f73d39e5.png">

<img width="1042" alt="Screen Shot 2022-01-06 at 16 27 45" src="https://user-images.githubusercontent.com/18718008/148370612-b0a9d4f2-0999-4e53-a0a1-16e8d6405a75.png">

<img width="1042" alt="Screen Shot 2022-01-06 at 16 28 57" src="https://user-images.githubusercontent.com/18718008/148370752-ee6ab09f-660b-4a45-a8ef-fb07defff0ce.png">

**Comment by jorgeorpinel** [comment]
Great idea. Maybe also "new in" (and "deprecated in"?) version notes to replace

![image](https://user-images.githubusercontent.com/1477535/149223695-84d2caed-7375-401c-83f6-7be5ec6c9161.png)

**Comment by jendefig** [comment]
I agree with @skshetry.  We should standardize this across docs and blogs.  I like this application.  Especially with the colors and emojis 😊 Maybe we should determine what categories they would be and define colors/emojis accordingly (preferably branded), like:

Warning = the light orange
Hints = the light blue
Notes = the light purple
? = the other light blue/purple/periwinkle?

This may be a better alternative to [the asides](https://github.com/iterative/dvc.org/issues/1117), because if the aside is long, it could make the whole page look wonky.  

But it might be nice to have them be collapsable like these so they are not taking up so much space? 
<img width="733" alt="image" src="https://user-images.githubusercontent.com/34289199/157752004-7f18d3e6-001a-40a2-a81a-f5b6444c19de.png">

**Comment by jendefig** [comment]
Then since we are getting all fancy, we should create a docs conventions page to go along with the Contributing page with these new cool functions.

**Comment by jorgeorpinel** [comment]
In docs we use Markdown block quotes for everything right now. The 3 categories I can think so from more to less common are:

1. General notes: the idea is to fade out the text to make them less visible bit still present. We may choose to add an emoji or not. They never have titles, it's just a paragraph usually.

2. Warnings: often we use block quotes now (not ideal since it fades out the text) but add ⚠️ to catch attention.

3. Tips 💡 and info. ℹ️  notes: They're basically the same thing except for the emoji so they can share a same design I think.

4. Separately we have the expandable `<details>` sections which contain details we want to have but hide by default. I guess the same component for general notes could just have a prop to determine if it's auto-hidden.

I have no idea what shapes and colors each should have, should we get a designer to propose something? Or find pre-existing designs from somewhere?

**Comment by daavoo** [comment]
> I have no idea what shapes and colors each should have, should we get a designer to propose something? Or find pre-existing designs from somewhere?

https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types ?

**Comment by jendefig** [comment]
> > I have no idea what shapes and colors each should have, should we get a designer to propose something? Or find pre-existing designs from somewhere?
> 
> https://squidfunk.github.io/mkdocs-material/reference/admonitions/#supported-types ?

These are lovely @daavoo! 😍

**Comment by julieg18** [comment]
For implementation, I can see two similiar options:

1. Use custom HTML elements. If we go this route, we'll probably have to wait on #3076 so we can create the same solution for both the blog and docs. 
Example:
```md
<block>

#### block title

text text text text text text text
<block>

```

2. Use an `<aside>` tag and rely on CSS. Spacy actually uses asides for their blocks. 
```md
<aside>

#### block title

text text text text text text text

<aside>
```

As for design, we can probably create a first iteration design based off the examples in this issue until @yalozhkin has a chance to take a look at it.

**Comment by rogermparent** [comment]
`aside`+CSS looks like a very appealing option, but turning `aside` into a full-width block instead of something to the side seems like it's breaking the spirit of the element.

There's also [a gatsby remark plugin for admonitions](https://www.gatsbyjs.com/plugins/gatsby-remark-admonitions/) with a syntax that's a little more remark-native and friendly, as well as coming with support for the standard admonitions out of the box:

Admonitions are a block element. 
The titles can include inline markdown and the body can include any block markdown except another admonition.

---

The general syntax is

```markdown
:::keyword optional title
some content
:::
```

For example,

```markdown
:::tip pro tip
`remark-admonitions` is pretty great!
:::
```


The default keywords are `important`, `tip`, `note`, `warning`, and `danger`.
Aliases for `info` => `important`, `success` => `tip`, `secondary` => `note` and `danger` => `warning` have been added for Infima compatibility.

---

A remark plugin like this would also work between docs and blog without/before #3076 (though I still support it)

**Comment by julieg18** [comment]
> There's also [a gatsby remark plugin for admonitions](https://www.gatsbyjs.com/plugins/gatsby-remark-admonitions/) with a syntax that's a little more remark-native and friendly, as well as coming with support for the standard admonitions out of the box:

Looks awesome, @rogermparent! The plugin looks very customizable and has everything we need! I'm all for using it vs creating a custom one from scratch!

**Comment by julieg18** [comment]
Attempted to use the plugin but got this weird error: 
<details>

```
 ERROR #85901  GRAPHQL

There was an error in your GraphQL query:

Cannot set properties of undefined (setting 'admonition')

   1 | query BlogPostPage($id: String!) {
   2 |   blogPost(id: {eq: $id}) {
   3 |     id
   4 |     slug
>  5 |     html
     |     ^
   6 |     timeToRead
   7 |     title
   8 |     date(formatString: "MMMM DD, YYYY")
   9 |     description
  10 |     descriptionLong
  11 |     tags
  12 |     commentsUrl
  13 |     author {
  14 |       name
  15 |       links {

File path: /Users/juliegalvan/Desktop/workspace/dvc.org/src/templates/blog-post.tsx
Url path: /blog/february-22-community-gems
Plugin: none



  TypeError: Cannot set properties of undefined (setting 'admonition')
  
  - index.js:214 Function.attacher
    [dvc.org]/[remark-admonitions]/lib/index.js:214:37
  
  - index.js:125 freeze
    [dvc.org]/[remark]/[unified]/index.js:125:31
  
  - index.js:263 Function.parse
    [dvc.org]/[remark]/[unified]/index.js:263:5
  
  - extend-node-type.js:197 parseString
    [dvc.org]/[gatsby-transformer-remark]/extend-node-type.js:197:34
  
  - extend-node-type.js:263 getMarkdownAST
    [dvc.org]/[gatsby-transformer-remark]/extend-node-type.js:263:14
  
  - extend-node-type.js:173 getAST
    [dvc.org]/[gatsby-transformer-remark]/extend-node-type.js:173:36
  
  - runMicrotasks
  
  - task_queues:96 processTicksAndRejections
    node:internal/process/task_queues:96:5
  
  - extend-node-type.js:357 getHTMLAst
    [dvc.org]/[gatsby-transformer-remark]/extend-node-type.js:357:21
  
  - extend-node-type.js:371 getHTML
    [dvc.org]/[gatsby-transformer-remark]/extend-node-type.js:371:21
  
  - index.js:20 
    [dvc.org]/[gatsby-plugin-parent-resolvers]/index.js:20:18
  
  - async Promise.all
  
  - async Promise.all
  
  - graphql-runner.ts:214 GraphQLRunner.query
    [dvc.org]/[gatsby]/src/query/graphql-runner.ts:214:14
  
  - query-runner.ts:135 queryRunner
    [dvc.org]/[gatsby]/src/query/query-runner.ts:135:14
```

</details>

Can't figure out the problem and apparently, this plugin has no stars or issues opened 😅 Any ideas, @rogermparent? Might need to complete #3076 first after all 🤔  

Besides that, here's what I've got for a first iteration design so far: 

<img width="717" alt="image" src="https://user-images.githubusercontent.com/43496356/158439826-e9962ac0-fcee-4d9e-97ea-f69eee620fb5.png">

I based the second one off of our details block and the first one more like the blockquotes. Any ideas for improvement are welcome!

**Comment by jorgeorpinel** [comment]
> Besides that, here's what I've got for a first iteration design

Looks better than what we do now but would be great to figure out the final format we should use in md to start rewriting docs.

**Comment by julieg18** [comment]
> Looks better than what we do now but would be great to figure out the final format we should use in md to start rewriting docs.

Now that #3076 is complete we can use custom HTML elements to create admonitions on blog and docs. 
Would look something like this:
```md
<admonition title"Note" type="note">

text text text...

</admonition>
```

Or we could work on a custom plugin that takes in `:::`:
```
:::tip pro tip
`remark-admonitions` is pretty great!
:::
```
But this will take longer... What do we prefer?

**Comment by jorgeorpinel** [comment]
Syntax-wise I vote for XML, which is native to md (i.e. `<admonition>`)
